File: app/src/main/java/org/wikipedia/analytics/OnThisDayFunnel.java
package org.wikipedia.analytics;

import androidx.annotation.NonNull;

import org.json.JSONObject;
import org.wikipedia.Constants;
import org.wikipedia.WikipediaApp;
import org.wikipedia.dataclient.WikiSite;

public class OnThisDayFunnel extends TimedFunnel {
    private static final String SCHEMA_NAME = "MobileWikiAppOnThisDay";
    private static final int REV_ID = 18118721;

    private final int source;
    private int maxScrolledPosition;

    public OnThisDayFunnel(WikipediaApp app, WikiSite wiki, Constants.InvokeSource source) {
        super(app, SCHEMA_NAME, REV_ID, Funnel.SAMPLE_LOG_ALL, wiki);
        this.source = source.ordinal();
    }

    @Override protected void preprocessSessionToken(@NonNull JSONObject eventData) { }

    public void scrolledToPosition(int position) {
        if (position > maxScrolledPosition) {
            maxScrolledPosition = position;
        }
    }

    public void done(int totalOnThisDayEvents) {
        log(
                "source", source,
                "totalEvents", totalOnThisDayEvents,
                "scrolledEvents", maxScrolledPosition
        );
    }
}

File: app/src/main/java/org/wikipedia/analytics/OnThisDayFunnel.kt
package org.wikipedia.analytics

import org.json.JSONObject
import org.wikipedia.Constants.InvokeSource
import org.wikipedia.WikipediaApp
import org.wikipedia.dataclient.WikiSite

class OnThisDayFunnel(app: WikipediaApp, wiki: WikiSite?, source: InvokeSource) :
        TimedFunnel(app, SCHEMA_NAME, REV_ID, SAMPLE_LOG_ALL, wiki) {

    private val source: Int = source.ordinal
    private var maxScrolledPosition = 0

    override fun preprocessSessionToken(eventData: JSONObject) {}

    fun scrolledToPosition(position: Int) {
        if (position > maxScrolledPosition) {
            maxScrolledPosition = position
        }
    }

    fun done(totalOnThisDayEvents: Int) {
        log(
                "source", source,
                "totalEvents", totalOnThisDayEvents,
                "scrolledEvents", maxScrolledPosition
        )
    }

    companion object {
        private const val SCHEMA_NAME = "MobileWikiAppOnThisDay"
        private const val REV_ID = 18118721
    }
}
